pkg/tests/cross-tests/create.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package crosstests
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tests/pulcheck"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+// Create validates that a Terraform provider witnesses the same input when:
+// - invoked directly with HCL on tfConfig
+// - bridged and invoked via Pulumi YAML on puConfig
+//
+// Create only applies to resources defined with github.com/hashicorp/terraform-plugin-sdk/v2. For cross-tests
+// on Plugin Framework based resources, see
+// github.com/pulumi/pulumi-terraform-bridge/pkg/pf/tests/internal/cross-tests.
+//
+// Create *does not* verify the outputs of the resource, only that the provider witnessed the same inputs.
+func Create(
+	t T, resource map[string]*schema.Schema, tfConfig cty.Value, puConfig resource.PropertyMap,
```

Comment:
Why do we need to expose both the pulumi and TF inputs in the test framework? I don't think it is the common use case for these to differ - should we instead aim to make it simpler to use this by having one input and handle the conversion in the framework, as before?

I don't think we have that many use cases for the inputs differing outside of type conversions as it this defeats the purpose of the test